rust-toolchain

```diff
@@ -7,14 +7,15 @@
 #   supported and used in CI.
 
 [toolchain]
-channel = "1.55.0"
+channel = "1.57.0"
 targets = [
     "aarch64-linux-android",
     "armv7-linux-androideabi",
     "i686-linux-android",
     "x86_64-linux-android",
     "aarch64-apple-ios",
     "x86_64-apple-ios",
+    "aarch64-apple-ios-sim"
```

Comment:
Exciting stuff! I believe this actually landed as part of the 1.56 work but happy for us not have to rely on nightly for our builds :). I'm a little concerned on the `ios-artifacts` failure given it's failing to build sqlcipher and nss

Reply:
Are you referring to this failure https://app.circleci.com/pipelines/github/mozilla/application-services/29255/workflows/9df293bb-df35-44b4-87e9-174ae44f8ee0/jobs/123114? or the `ios-artifacts` failing on main - the latter should be fixed by this but the sqlcipher build failure on the `cargo bench` that this PR is introducing thou... I'll have to look into it a little

Reply:
ah didn't realize there was also a main one, I was looking specifically at this sqlcipher one since it seems a bunch of sqlcipher artifacts go missing. Once I free up some time I might help look as well!

Reply:
Yeah the one on main is a known rust nightly bug that should get fixed soon, and it will also get fixed with this PR (since we move away from using nightly 🎉) the sqlcipher one I haven't looked into, but it's weird that `cargo test` works, but `cargo bench` causes trouble... and I can't repro locally either 🙃